### PR TITLE
[improve](Nereids): enhance equals UT

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

- enhance the equals UT
- add EqualTo of
- minor

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/test/java/org/apache/doris/nereids/trees/plans/PlanEqualsTest.java [hunk 1]
```diff
@@ -48,193 +49,259 @@
...
     public void testLogicalAggregate(@Mocked Plan child) {
-        List<Expression> groupByExprList = Lists.newArrayList();
         List<NamedExpression> outputExpressionList = ImmutableList.of(
-                new SlotReference("a", new BigIntType(), true, Lists.newArrayList())
-        );
-
-        LogicalAggregate one = new LogicalAggregate(groupByExprList, outputExpressionList, child);
-        Assertions.assertEquals(one, one);
+                new SlotReference(new ExprId(0), "a", new BigIntType(), true, Lists.newArrayList()));
+        LogicalAggregate<Plan> logicalAggregate = new LogicalAggregate<>(Lists.newArrayList(), outputExpressionList,
```
Comment (morrySnow) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/trees/plans/PlanEqualsTest.java:61]: we could use `actual`, `expected` and `unexpected` as their names

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/EqualTo.java [hunk 2]
```diff
@@ -33,6 +33,10 @@ public EqualTo(Expression left, Expression right) {
         super(left, right, "=");
     }
 
+    public static EqualTo of(Expression left, Expression right) {
```
Comment (adonis0147) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/EqualTo.java:36]: What's the benefit of introducing the static method `EqualTo::of`?

Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/EqualTo.java:36]: Like ImmutableList.of()
we don,t need new

Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/EqualTo.java:36]: I think it is not evil to use `new` to instantiate an object. JVM may not always inline some methods. An extra wrapper method which just passes argument to `new` only may introduce some costs.

**Review by morrySnow — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**